### PR TITLE
improved the speed comparison to reduce noise

### PR DESCRIPTION
**Pull request by tarekziade**
run it 3 times and get the min

### CODE REVIEW

File: connectors/tests/test_utils.py [hunk 1]
```diff
@@ -58,21 +60,42 @@ def test_invalid_names():
...
+
+    async def run():
+        for i in range(1000):
+            await queue.put("x" * 100)
+
+    asyncio.run(run())
+    """
+    )
+
+    mem_queue_duration = min(timeit.repeat(memqueue_script, number=1, repeat=3))
```
Comment (timgrein) [inline connectors/tests/test_utils.py:81]:
```suggestion

     def mem_queue():
        import asyncio
        from connectors.utils import MemQueue

        queue = MemQueue(maxmemsize=1024 * 1024, refresh_interval=0.1, refresh_timeout=2)

        async def run():
            for i in range(1000):
                await queue.put("x" * 100)

        asyncio.run(run())

    mem_queue_duration = min(timeit.repeat(mem_queue, number=1, repeat=3))
```

I think we should use nested functions as we lose auto-suggestions with `dedent` in IDEs.

File: connectors/tests/test_utils.py [hunk 2]
```diff
@@ -58,21 +60,42 @@ def test_invalid_names():
...
+
+    async def run():
+        for i in range(1000):
+            await queue.put("x" * 100)
+
+    asyncio.run(run())
+    """
+    )
+
+    queue_duration = min(timeit.repeat(queue_script, number=1, repeat=3))
```
Comment (timgrein) [inline connectors/tests/test_utils.py:98]:
```suggestion

     def vanilla_queue():
        import asyncio

        queue = asyncio.Queue()

        async def run():
            for i in range(1000):
                await queue.put("x" * 100)

        asyncio.run(run())

    queue_duration = min(timeit.repeat(vanilla_queue, number=1, repeat=3))
```

Same point as above.

**Review by timgrein — COMMENTED** [review]
Nice, that we're now using `timeit`. Added two comments, where I would suggest that we use nested functions instead of `dedent` as we're loosing auto-suggestions/highlighting etc. in IDEs.

File: connectors/tests/test_utils.py [hunk 3]
```diff
@@ -58,21 +60,42 @@ def test_invalid_names():
...
-async def test_mem_queue_speed(patch_logger):
+def test_mem_queue_speed(patch_logger):
     # with memqueue
+
+    memqueue_script = dedent(
+        """
+    import asyncio
+    from connectors.utils import MemQueue
+
     queue = MemQueue(maxmemsize=1024 * 1024, refresh_interval=0.1, refresh_timeout=2)
```
Comment (timgrein) [inline connectors/tests/test_utils.py:71]:
```suggestion

```

Comment (timgrein) [inline connectors/tests/test_utils.py:71]:
```suggestion

```

File: connectors/tests/test_utils.py [hunk 4]
```diff
@@ -58,21 +60,42 @@ def test_invalid_names():
...
+    )
+
+    mem_queue_duration = min(timeit.repeat(memqueue_script, number=1, repeat=3))
 
     # vanilla queue
+    queue_script = dedent(
+        """
+    import asyncio
+
     queue = asyncio.Queue()
```
Comment (timgrein) [inline connectors/tests/test_utils.py:88]:
```suggestion

```

**Comment by timgrein** [comment]
Oops, my mousepad is super sensitive, accidentally pressed update haha

**Review by timgrein — APPROVED** [review]
LGTM, still not a fan of `dedent` vs. a nested function, but using `timeit` looks good.

**Comment by tarekziade** [comment]
> LGTM, still not a fan of `dedent` vs. a nested function (from an IDE user perspective), but using `timeit` looks good. Thanks for the change.

erm, I thought I applied your proposed changes, will try again

**Pull request merged**